File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector < vector < string >> out;
        map < map < char, int > , vector < string >> dict; //key - value
        for (string str: strs) {
            map < char, int > m;
            for (int i = 0; i < str.size(); i++) {
                m[str[i]]++;
            }
            dict[m].push_back(str);
        }
        for (auto x: dict) {
            out.push_back(x.second);
        }
        return out;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var dic = new Dictionary<string, IList<string>>();

        foreach (string str in strs)
        {
            int[] count = new int[26];
            foreach (char c in str)
            {
                count[c - 'a']++;
            }
            string key = string.Join(",",count);
            if (!dic.ContainsKey(key))
            {
                
                dic[key] = new List<string>();
            }
            dic[key].Add(str);
        }
        return new List<IList<string>>(dic.Values);
    }
}